wgpu-core/src/track/assertions.rs

```diff
@@ -0,0 +1,49 @@
+//! Macros for validation internal to the resource tracker.
+//!
+//! This module defines assertion macros that respect `wgpu-core`'s
+//! `"internal_asserts"` feature.
+//!
+//! Because `wgpu-core`'s public APIs validate their arguments in all
+//! types of builds, for performance, the `track` module skips some of
+//! Rust's usual run-time checks on its internal operations in release
+//! builds. However, some `wgpu-core` applications have a strong
+//! preference for robustness over performance. To accommodate them,
+//! `wgpu-core`'s `"internal_asserts"` feature enables that validation
+//! in both debug and release builds.
+
+#[cfg(feature = "internal_asserts")]
+macro_rules! tracker_assert {
```

Comment:
Need to change the macro/function names too

Reply:
*sigh* thanks